aws/resource_aws_lb_listener_rule_test.go

```diff
@@ -829,3 +879,190 @@ resource "aws_security_group" "alb_test" {
+
+resource "aws_vpc" "alb_test" {
+  cidr_block = "10.0.0.0/16"
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}
+
+resource "aws_subnet" "alb_test" {
+  count                   = 2
+  vpc_id                  = "${aws_vpc.alb_test.id}"
+  cidr_block              = "${element(var.subnets, count.index)}"
+  map_public_ip_on_launch = true
+  availability_zone       = "${element(data.aws_availability_zones.available.names, count.index)}"
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}
+
+resource "aws_security_group" "alb_test" {
+  name        = "allow_all_alb_test"
+  description = "Used for ALB Testing"
+  vpc_id      = "${aws_vpc.alb_test.id}"
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags {
+    Name = "TestAccAWSALB_basic"
+  }
+}`, lbName, targetGroupName)
+}
+
+func testAccAWSLBListenerRuleConfig_autoPriorityFirst(lbName, targetGroupName string) string {
+	return testAccAWSLBListenerRuleConfig_autoPriorityBase(lbName, targetGroupName) + fmt.Sprintf(`
+resource "aws_lb_listener_rule" "first" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+
+  action {
+    type = "forward"
+    target_group_arn = "${aws_lb_target_group.test.arn}"
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/first/*"]
+  }
+}
+
+resource "aws_lb_listener_rule" "third" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+  priority = 3
+
+  action {
+    type = "forward"
+    target_group_arn = "${aws_lb_target_group.test.arn}"
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/third/*"]
+  }
+}
+`)
+}
+
+func testAccAWSLBListenerRuleConfig_autoPriorityLast(lbName, targetGroupName string) string {
+	return testAccAWSLBListenerRuleConfig_autoPriorityFirst(lbName, targetGroupName) + fmt.Sprintf(`
+resource "aws_lb_listener_rule" "last" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+
+  action {
+    type = "forward"
+    target_group_arn = "${aws_lb_target_group.test.arn}"
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/last/*"]
+  }
+}
+`)
+}
+
+func testAccAWSLBListenerRuleConfig_autoPriorityStatic(lbName, targetGroupName string) string {
+	return testAccAWSLBListenerRuleConfig_autoPriorityFirst(lbName, targetGroupName) + fmt.Sprintf(`
+resource "aws_lb_listener_rule" "last" {
+  listener_arn = "${aws_lb_listener.front_end.arn}"
+  priority = 7
+
+  action {
+    type = "forward"
+    target_group_arn = "${aws_lb_target_group.test.arn}"
+  }
+
+  condition {
+    field = "path-pattern"
+    values = ["/last/*"]
+  }
+}
+`)
+}
+
+func testAccAWSLBListenerRuleConfig_autoPriority100Rules(lbName, targetGroupName string) string {
+	return testAccAWSLBListenerRuleConfig_autoPriorityStatic(lbName, targetGroupName) + fmt.Sprintf(`
+resource "aws_lb_listener_rule" "limit" {
+  count = 97
```

Comment:
Can we make this match the test name or vice versa please?

Reply:
Sorry I'm not sure if I get your point. Did you mean either `count = 100` or `testAccAWSLBListenerRuleConfig_autoPriority97Rules`? If that's the case, maybe change to `testAccAWSLBListenerRuleConfig_autoPriorityRuleNumberLimit`? The intension was to fill the listener with 100 rules limit per load balancer and there was already 3 existing rules.

Reply:
Okay I wasn't immediately aware of the rules per listener limit.

If we're just trying to test concurrency of auto-assigning priorities (I think more important here), let's just do 10, which should always pass.

If we're trying to test hitting the max rules per listener limit (I do not think its important to test one below the limit), that should be a separate test of 101 rules and `ExpectError`. AWS could change that limit and it makes it hard to understand why we're testing 3 + 97 rules here without making the test name more explicit like you suggested with `testAccAWSLBListenerRuleConfig_autoPriorityRuleNumberLimit`.

Reply:
I'll do both then, testing for 10 and 101.